analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -375,52 +376,63 @@ public void Finally_WrappedInLocalLifetimeRegion()
         public void Finally_ThrowInTry()
         {
             const string code = @"
-Tag(""BeforeTry"");
+var tag = ""BeforeTry"";
 try
 {
     Tag(""InTry"");
     throw new System.Exception();
-    Tag(""UnreachableInTry"");
+    tag = ""UnreachableInTry"";
 }
 finally
 {
-    Tag(""InFinally"");
+    tag = ""InFinally"";
 }
-Tag(""UnreachableAfterFinally"");";
-            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+tag = ""UnreachableAfterFinally"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
                 "BeforeTry",
                 "InTry",
-                "InFinally");
+                "InFinally",                // With Exception thrown by Tag("InTry")
+                "InFinally");               // With Exception thrown by `throw`
+
+            validator.TagStates("InFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
         }
 
         [TestMethod]
         public void Finally_ThrowInFinally()
         {
             const string code = @"
-Tag(""BeforeTry"");
+var tag = ""BeforeTry"";
```

Comment:
_Open question:_ Do we need to throw on line 410 for purpose of this test `Finally_ThrowInFinally`? The finally would be visited with happy path too.